channeldb/channel.go

```diff
@@ -389,6 +395,12 @@ type OpenChannel struct {
 	sync.RWMutex
 }
 
+// SetResolver sets resolveTCP to a resolver other than the default
```

Comment:
This isn't currently used in the PR. I now remember it's original purpose: to ensure that any DNS resolutions we performed underlying deserialization didn't also leak to the clearnet if we were in tor mode. I think a better solution is to _remove any sort of DNS queries_ during deserialization all together. This would mean modifying the on-disk storage format of the addresses to _not_ just store the string, but store the raw representation. 

So in short, we abstract out [this logic](https://github.com/lightningnetwork/lnd/blob/master/channeldb/graph.go#L1740), so it can be used for both `LinkNodes` as well as `LightningNodes`.